### PR TITLE
Better multiline performance

### LINKED ISSUE #80: Bad performance on arguments in new lines
```
a b c d e f g h i j k l m
```
Compiles instantly
while
```
a 
  b 
  c
  d
  e
  f
  g
  h
  i
  j
  k
  l
  m
```
Would take couple of minutes

### PR DESCRIPTION
**Pull request by baransu**
Attempt to fix #80.
Updates also tests to newest `elm-test` version.

Main optimisation was reordering `term` and `expression` order based on popularity. It reduced time a lot. Other changes were related to creating more GC friendly Elm because this is the real problem. There is test in `tests/Multiline.elm` which parses 11 arguments, every in new line. It takes around 20s on MacBook Pro (down from ~96s). I don't thing we can squash more than that without changing new line parsing logic.

Please try to add `Test.only` to `tests/Multiline.elm` to run only performance test and measure parsing time. Really counting on your feedback.

### CODE REVIEW

File: example/elm.js [hunk 1]
```diff
@@ -5404,16 +5404,20 @@ var _Bogdanp$elm_ast$Ast_Helpers$reservedOperators = {
 		}
 	}
```
Comment (wende) [inline example/elm.js:5405]: This file shouldn't be here

File: tests/Multiline.elm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                            )
+                                            (var "g")
+                                        )
+                                        (var "h")
+                                    )
+                                    (var "i")
+                                )
+                                (var "j")
+                            )
+                            (var "k")
```
Comment (wende) [inline tests/Multiline.elm]: XD

Comment (wende) [inline tests/Multiline.elm]: 🤣 
Could you reduce it to something less hideous please?
Like for example
```
makeApplication : Expression -> List Expression -> List Expression
makeApplication = List.foldr (flip Application)
[...]
|> is makeApplication (var "fn") [var "a", var "b", var "c", var "d" ...., var "k"]
```

File: tests/Main.elm [hunk 3]
```diff
@@ -1,23 +0,0 @@
...
-
-port emit : ( String, Value ) -> Cmd msg
-
-
-all : Test
-all =
-    describe "Ast suite"
-        [ Expression.all
-        , Statement.all
-        ]
```
Comment (wende) [inline tests/Main.elm:23]: Where are the tests executed now though?

**Review by wende — CHANGES_REQUESTED** [review]
Great job overall 👍 
Few comments from me

File: tests/Main.elm (hunk 3 above)
Comment (baransu) [reply tests/Main.elm:23]: `elm-test` executes all exposed tests from test files.

File: example/elm.js (hunk 1 above)
Comment (baransu) [reply example/elm.js:5405]: `example/elm.js` have to be in the repo because of http://bogdanp.github.io/elm-ast/example/

Comment (wende) [reply example/elm.js:5405]: @Baransu it's only updated on branch `gitpages`

**Pull request closed without merging**